src/OpenLoco/Ui/Screenshot.cpp

```diff
@@ -117,7 +90,117 @@ namespace OpenLoco::Input
+                break;
+            }
+
+            fileName = std::string(scenarioName) + " (" + std::to_string(suffix) + ").png";
+        }
+
+        if (path.empty())
+        {
+            throw std::runtime_error("Failed finding filename");
+        }
+
+        std::fstream outputStream(path.c_str(), std::ios::out | std::ios::binary);
+        saveRenderTargetToPng(rt, outputStream);
 
         return fileName;
     }
+
+    std::string saveScreenshot()
+    {
+        auto& rt = Gfx::getScreenRT();
+        return prepareSaveScreenshot(rt);
+    }
+
+    std::string saveGiantScreenshot()
+    {
+        const auto& main = WindowManager::getMainWindow();
+        const auto zoomLevel = main->viewports[0]->zoom;
+
+        const uint16_t resolutionWidth = ((Map::kMapColumns * 32 * 2) >> zoomLevel) + 8;
+        const uint16_t resolutionHeight = ((Map::kMapRows * 32 * 1) >> zoomLevel) + 128;
+
+        Ui::Viewport viewport{};
+        viewport.width = resolutionWidth;
+        viewport.height = resolutionHeight;
+        viewport.x = 0;
+        viewport.y = 0;
+        viewport.viewWidth = viewport.width;
+        viewport.viewHeight = viewport.height;
+        viewport.zoom = zoomLevel;
+        viewport.pad_11 = 0;
+        viewport.flags = 0;
+
+        const uint16_t centreX = (Map::kMapColumns / 2) * 32 + 16;
+        const uint16_t centreY = (Map::kMapRows / 2) * 32 + 16;
+
+        const uint16_t z = Map::TileManager::getHeight({ centreX, centreY }).landHeight;
+        const auto rotation = main->viewports[0]->getRotation();
+        uint16_t x = 0, y = 0;
+
+        // TODO: I think we have a function for this already?
+        switch (rotation)
+        {
+            case 0:
+                x = centreY - centreX;
+                y = ((centreX + centreY) / 2) - z;
+                break;
+            case 1:
+                x = -centreY - centreX;
+                y = ((-centreX + centreY) / 2) - z;
+                break;
+            case 2:
+                x = -centreY + centreX;
+                y = ((-centreX - centreY) / 2) - z;
+                break;
+            case 3:
+                x = centreY + centreX;
+                y = ((centreX - centreY) / 2) - z;
+                break;
+        }
+
+        viewport.viewX = x - ((viewport.viewWidth << zoomLevel) / 2);
+        viewport.viewY = y - ((viewport.viewHeight << zoomLevel) / 2);
+
+        // Ensure sprites appear regardless of rotation
+        // TODO: what is the OpenLoco equivalent?
+        // reset_all_sprite_quadrant_placements();
+
+        Gfx::RenderTarget rt{};
+        rt.bits = static_cast<uint8_t*>(malloc(resolutionWidth * resolutionHeight));
+        if (!rt.bits)
+            return nullptr;
+
+        rt.x = 0;
+        rt.y = 0;
+        rt.width = resolutionWidth;
+        rt.height = resolutionHeight;
+        rt.pitch = 0;
+        rt.zoomLevel = 0;
+
+        viewport.render(&rt);
+        free(rt.bits);
```

Comment:
use after free! You need to prepareSaveScreenshot then free the memory not the other way round!

Reply:
Argh! Thanks for spotting that, I'd been looking at this way too long.